packages/inferno/src/index.ts

```diff
@@ -35,27 +35,29 @@ import { createRef, forwardRef, mountRef } from './core/refs';
 export * from './core/types';
 
 if (process.env.NODE_ENV !== 'production') {
-  // Checks if Inferno is running in jest testing environment.
-  const testingEnv =
-    typeof process === 'object' && process.env?.JEST_WORKER_ID !== undefined;
```

Comment:
I think we should keep this process.env?.JEST_WORKER_ID check here also to avoid breaking change for jest users

Reply:
I can do that, but keep in mind, what i wrote in the commit message:

> FYI: JEST related warning skip about development mode was removed with https://github.com/infernojs/inferno/commit/2ed75dc9e8d8dfa0d77042c31f93cf0e50806720 when fixing https://github.com/infernojs/inferno/issues/1582, only the skipping of minified code warning was in place

Shall I fix that, and add it back?

Reply:
hmm sorry I don't get it, what do you mean?

If the test framework is running against minified code then it should also define process.env.NODE_ENV production, otherwise do not minify the code and then process.env.NODE_ENV does not need to be set and no warning is produced

Reply:
I agree with the functionality (how it should be), I'm just saying, that on the current `master` branch, if you are running the non-minified `dist`-version of `inferno` in `jest`-suite, you will see the warning (that inferno is running in development mode) shown as console log, because the code was incorrectly fixed in #1582 with commit 2ed75dc9e8d8dfa, as this happened:

```
-if (process.env.NODE_ENV !== 'production' && !testingEnv) {
   /* tslint:disable-next-line:no-empty */
   const testFunc = function testFn() {};
   /* tslint:disable-next-line*/
   console.log('Inferno is in development mode.');
 
-  if (((testFunc as Function).name || testFunc.toString()).indexOf('testFn') === -1) {
+  if (!testingEnv && ((testFunc as Function).name || testFunc.toString()).indexOf('testFn') === -1) {
```

The `console.log('Inferno is in development mode.');` instruction got moved out from the `if !testingEnv` branch.

Anyways, I've added it back, see new diff.